expertise/service/utils.py

```diff
@@ -158,3 +159,166 @@ def get_user_id(openreview_client):
+        self.user_id = None
+        self.job_id = None
+        self.token = starting_config.get('token', None)
+        self.baseurl = starting_config.get('baseurl', None)
+        self.baseurl_v2 = starting_config.get('baseurl_v2', None)
+        self.dataset = starting_config.get('dataset', {})
+
+        # Optional fields
+        self.model = starting_config.get('model', None)
+        self.exclusion_inv = starting_config.get('exclusion_inv', None)
+        self.paper_invitation = starting_config.get('paper_invitation', None)
+        self.paper_id = starting_config.get('paper_id', None)
+
+        # Optional model params
+        self.model_params = starting_config.get('model_params', {})
+        model_params = starting_config.get('model_params', {})
+        self.model_params = {}
+        self.model_params['use_title'] = model_params.get('use_title', None)
+        self.model_params['use_abstract'] = model_params.get('use_abstract', None)
+        self.model_params['average_score'] = model_params.get('average_score', None)
+        self.model_params['max_score'] = model_params.get('max_score', None)
+        self.model_params['skip_specter'] = model_params.get('skip_specter', None)
+        self.model_params['batch_size'] = model_params.get('batch_size', 1)
+        self.model_params['skip_cuda'] = model_params.get('skip_cuda', False)
+    
+    def _get_required_field(self, req, superkey, key):
+        try:
+            field = req[key]
+        except KeyError:
+            raise openreview.OpenReviewException(f"Bad request: required field missing in {superkey}: {key}")
+        return field
+
+    def _load_entity(self, entity_id, entity):
+        '''Load information from an entity into the config'''
+        def _get_from_entity(key):
+            return self._get_required_field(entity, entity_id, key)
+
+        type = _get_from_entity('type')
+        # Handle type group
+        if type == 'Group':
+            if 'memberOf' in entity.keys():
+                members_group = _get_from_entity('memberOf')
+                exc_inv = entity.get('expertise', {}).get('exclusion', {}).get('invitation', None)
+
+                self.match_group = members_group
+                self.exclusion_inv = exc_inv
+            else:
+                raise openreview.OpenReviewException(f"Bad request: no valid {type} properties in {entity_id}")
+        # Handle type note
+        elif type == 'Note':
+            if 'invitation' in entity.keys():
+                inv = _get_from_entity('invitation')
+                self.paper_invitation = inv
+            elif 'id' in entity.keys():
+                id = _get_from_entity('id')
+                self.paper_id = id
+            else:
+                raise openreview.OpenReviewException(f"Bad request: no valid {type} properties in {entity_id}")
+        else:
+            raise openreview.OpenReviewException(f"Bad request: invalid type in {entity_id}")
+
+    def from_request(self, request):
```

Comment:
why don't you call it `from_json`?

Reply:
I called it `from_request` because the dict that it expects has the new API format (camel case fields), where as the `to_json` function converts the class into a dict that follows a format that is accepted by the expertise model (underscore/snake fields). I made this distinction to allow for a `from_json` function that is able to accept a dict formatted for the expertise model.

However, I'm open to changing it to `from_json` since I think this class will only be used for validation and loading from the initial API call.